cpp/src/arrow/compute/kernel.cc

```diff
@@ -445,11 +445,13 @@ const TypeMatcher& InputType::type_matcher() const {
 
 Result<TypeHolder> OutputType::Resolve(KernelContext* ctx,
                                        const std::vector<TypeHolder>& types) const {
-  if (kind_ == OutputType::FIXED) {
-    return type_.get();
-  } else {
-    return resolver_(ctx, types);
+  switch (kind_) {
+    case OutputType::FIXED:
+      return type_;
+    case OutputType::COMPUTED:
+      break;
   }
+  return resolver_(ctx, types);
```

Comment:
How about returning an error by default?

```suggestion
      return resolver_(ctx, types);
  }
  return Status::NotImplemented("Must not happen"); // UnknownError?
```

Reply:
Overkill. That `return Status` would only ever be reached if `kind_` is poisoned memory: in that case all that we need is consistent behavior to not go crazy debugging.